### PR TITLE
Initial work for OpenShift Marketplace

### PR DESCRIPTION
**Pull request by disposab1e**
Hi John,

after so many quickstarts it's time to quickstart :-)

This is the initial work to install the operator through OpenShift Marketplace. It's just a beginning but it works for me in OpenShift 4.2 (crc).

I have a port for Kubernetes and it's running fine but the operator is not displayed in the OLM console. I need to check is it a problem with my csv or the console.

Please see the README.md files for more information.

### CODE REVIEW

**Comment by jmckind** [comment]
Reviewing this and testing locally, thanks @disposab1e!

**Comment by disposab1e** [comment]
cool! I'm not sure to add the "Application" and "AppProject" API's at this point. Perhaps just for viewing existing instances of this objects and not for adding/changing.

**Comment by jmckind** [comment]
Yeah, I cheated a bit and pulled those in from the main argo-cd repo and don't feel like that the is right approach to be honest, as maintaining two copies is not ideal.

File: marketplace/openshift/bundle/argocd-operator.v0.0.1.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      - kind: Application
+        name: applications.argoproj.io
+        version: v1alpha1
+        displayName: Application
+        description: 'Represents a definition of Application resource.'
+      - kind: AppProject
+        name: appprojects.argoproj.io
+        version: v1alpha1
+        displayName: AppProject
+        description: 'Represents a logical grouping of applications.'
```
Comment (disposab1e) [inline marketplace/openshift/bundle/argocd-operator.v0.0.1.clusterserviceversion.yaml]: so better we remove them from here?

**Comment by disposab1e** [comment]
and delete the files from the bundle?
marketplace/openshift/bundle/argoproj_v1alpha1_application_crd.yaml
marketplace/openshift/bundle/argoproj_v1alpha1_appproject_crd.yaml

**Comment by jmckind** [comment]
Yeah probably best, but need to think through how to handle it. We can keep it in there for now if that keeps things working for you and I still need to test all of this on my end.

Ideally if the operator itself is part of the main argo-cd repo, this is not a problem. Having those CRDs in this repo has the effect of pinning a version of the operator to a specific version of the CRDs and we will need to ensure any changes get applied here, if that makes sense.

I would like to not do that if possible, and have the operator support upgrading to different versions of ArgoCD for example.

**Comment by disposab1e** [comment]
No problem to keep everything untouched... and yes agree makes no sense to pin the operator to a specific version of the crd. but to be honest... I need to learn more about "the same operator version can handle differnet versions of crd's" ... but I will do :-)

**Comment by jmckind** [comment]
Going to merge this in for now and we can iterate on changes with further testing.

**Pull request merged**